### DISCUSSION TITLE
Having trouble with install

### DISCUSSION THREAD

**Discussion by mkelley88** [blakeblackshear/frigate#2272] [Ask A Question] [answered]
Hello, 

I am trying to get Frigate running on TrueNAS SCALE. I have got it all set up, config and all, but it is experiencing a timeout and exits. It won't stay running and I'm not sure how to correct this. I recently set up Frigate inside Home Assistant (hassos) and the config worked good on there.

In my logs (dmesg) I get the following:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited [0.]
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2021-11-17 09:31:33] frigate.app                    INFO    : Starting Frigate (0.9.4-26ae608)
[2021-11-17 09:31:33] frigate.app                    INFO    : Creating directory: /tmp/cache
Starting migrations
[2021-11-17 09:31:33] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2021-11-17 09:31:33] peewee_migrate                 INFO    : There is nothing to migrate
timed out
[cmd] python3 exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
```

**Comment by blakeblackshear** [comment, marked as answer]
I believe this is a timeout for connecting to your mqtt server.

**Reply by mkelley88** [reply]
Thank you! This did it. I had my authentication set for my Home Assistant based MQTT and apparently something broke. I fired up a new Mosquitto server and moved it to that one and it's working fine now. I DID have to add a Memory Backed Volume of "/dev/shm" to my TrueNAS config, because of bus errors (for those who read this and run in to that same problem).

**Comment by mitchross** [comment]
@mkelley88  how did you get this running in True Nas scale? With helm chart? Or pure docker image? If the latter how did you add a config file?

**Reply by mkelley88** [reply]
I used the Truecharts Custom app template. It's sketchy, but it did work for awhile. Something happened recently and it stopped working for me and I haven't got around to trying to fix it. Honestly Kubernetes seems so complicated to set up and use (at least without a proper chart), but then again I am just now still trying to get a deeper understanding of Docker too. I don't understand docker-compose either yet. I'll leave you screenshots of my config. It worked great when it worked though. For the config file and the data location I just added host path back to a truenas dataset directory. 


<img width="764" alt="Screen Shot 2022-01-02 at 9 47 48 AM" src="https://user-images.githubusercontent.com/5567324/147879772-b5cda19e-1a7d-4a33-9e8d-395f99a9b4da.png">
<img width="749" alt="pass env var" src="https://user-images.githubusercontent.com/5567324/147892180-210ec4e5-4085-45db-a2ba-bbd9e756f0b3.png">
<img width="738" alt="Screen Shot 2022-01-02 at 9 48 25 AM" src="https://user-images.githubusercontent.com/5567324/147879775-ad540727-60e0-4c86-97f5-aaa1cbc101ae.png">
<img width="749" alt="Screen Shot 2022-01-02 at 9 48 44 AM" src="https://user-images.githubusercontent.com/5567324/147879776-867d7999-219a-442b-9745-96075ab42b50.png">
<img width="736" alt="Screen Shot 2022-01-02 at 9 49 01 AM" src="https://user-images.githubusercontent.com/5567324/147879777-d9922894-66f9-4f37-96aa-d63f0f4b5467.png">
<img width="746" alt="Screen Shot 2022-01-02 at 9 49 17 AM" src="https://user-images.githubusercontent.com/5567324/147879778-88cd7d2f-1c83-4be6-ab29-9002f3e0355c.png">

**Reply by mitchross** [reply]
Thanks! I fully understand docker-compose. I did some research, True Nas scale uses k3s for "app" deployment and its locked down. Theres no way to do native docker/docker-compose, so you are forced to do the janky way you described. Theres a pretty entertaining thread on TrueNas's screw up https://www.truenas.com/community/threads/container-virtualization-and-the-scale-rc-1-reality.97137/

**Reply by mkelley88** [reply]
Pretty much, yes. In addition to having to fiddle around with just getting Frigate running, I also found out that my TrueNAS SCALE VMs cannot communicate with other services on the same host. For instance, I have Home Assistant running in one VM and Ubuntu (for easier Docker containers, Portainer, etc) in a second VM. If I am inside either one of the VMs and attempt to ping my TrueNAS host it fails. I can reach any other address on my network, including the internet, just not the host IP. Because they can't reach TrueNAS, I cannot use network file sharing, I can't have Frigate connected to Home Assistant. One of my whole objectives for switching to SCALE is so I could stop running my security software on a Windows VM. Built in Docker support was what I was hoping for. They failed. I don't need k3s, I just need an easy to use Docker option Alas, I must now build all my docker stuff inside a VM. :(
